### ISSUE TITLE
Add tests for scripts like train_hypergrid with replay_buffer > 0

### ISSUE THREAD

**Issue by saleml** [GFNOrg/torchgfn#167]
A recent PR introduced a bug where train_hypergrid doesn't converge if replay_buffer is set to 100.

This should be caught with the tests !

**Comment by hyeok9855** [comment]
Yes, of course. I also wanted to check the buffer side. I planned to add an example that uses the replay buffer for off-policy training. I will also check the tests!

**Comment by saleml** [comment]
Done in #276
